include/dgl/aten/array_ops.h

```diff
@@ -0,0 +1,254 @@
+ * \param nbits result array's bits (32 or 64)
+ * \param ctx Device context
+ * \return the result array
+ */
+IdArray Full(int64_t val, int64_t length, uint8_t nbits, DLContext ctx);
+
+/*! \brief Create a deep copy of the given array */
+IdArray Clone(IdArray arr);
+
+/*! \brief Convert the idarray to the given bit width */
+IdArray AsNumBits(IdArray arr, uint8_t bits);
+
+/*! \brief Arithmetic functions */
+IdArray Add(IdArray lhs, IdArray rhs);
+IdArray Sub(IdArray lhs, IdArray rhs);
+IdArray Mul(IdArray lhs, IdArray rhs);
+IdArray Div(IdArray lhs, IdArray rhs);
+
+IdArray Add(IdArray lhs, int64_t rhs);
+IdArray Sub(IdArray lhs, int64_t rhs);
+IdArray Mul(IdArray lhs, int64_t rhs);
+IdArray Div(IdArray lhs, int64_t rhs);
+
+IdArray Add(int64_t lhs, IdArray rhs);
+IdArray Sub(int64_t lhs, IdArray rhs);
+IdArray Mul(int64_t lhs, IdArray rhs);
+IdArray Div(int64_t lhs, IdArray rhs);
+
+IdArray Neg(IdArray array);
+
+// XXX(minjie): currently using integer array for bool type
+IdArray GT(IdArray lhs, IdArray rhs);
+IdArray LT(IdArray lhs, IdArray rhs);
+IdArray GE(IdArray lhs, IdArray rhs);
+IdArray LE(IdArray lhs, IdArray rhs);
+IdArray EQ(IdArray lhs, IdArray rhs);
+IdArray NE(IdArray lhs, IdArray rhs);
+
+IdArray GT(IdArray lhs, int64_t rhs);
+IdArray LT(IdArray lhs, int64_t rhs);
+IdArray GE(IdArray lhs, int64_t rhs);
+IdArray LE(IdArray lhs, int64_t rhs);
+IdArray EQ(IdArray lhs, int64_t rhs);
+IdArray NE(IdArray lhs, int64_t rhs);
+
+IdArray GT(int64_t lhs, IdArray rhs);
+IdArray LT(int64_t lhs, IdArray rhs);
+IdArray GE(int64_t lhs, IdArray rhs);
+IdArray LE(int64_t lhs, IdArray rhs);
+IdArray EQ(int64_t lhs, IdArray rhs);
+IdArray NE(int64_t lhs, IdArray rhs);
+
+/*! \brief Stack two arrays (of len L) into a 2*L length array */
+IdArray HStack(IdArray arr1, IdArray arr2);
+
+/*!
+ * \brief Return the data under the index. In numpy notation, A[I]
+ * \tparam ValueType The type of return value.
+ */
+template<typename ValueType>
+ValueType IndexSelect(NDArray array, uint64_t index);
+NDArray IndexSelect(NDArray array, IdArray index);
+
+/*!
+ * \brief Permute the elements of an array according to given indices.
+ *
+ * Equivalent to:
+ *
+ * <code>
+ *     result = np.zeros_like(array)
+ *     result[indices] = array
+ * </code>
+ */
+NDArray Scatter(NDArray array, IdArray indices);
+
+/*!
+ * \brief Repeat each element a number of times.  Equivalent to np.repeat(array, repeats)
+ * \param array A 1D vector
+ * \param repeats A 1D integer vector for number of times to repeat for each element in
+ *                \c array.  Must have the same shape as \c array.
+ */
+NDArray Repeat(NDArray array, IdArray repeats);
+
+/*!
+ * \brief Relabel the given ids to consecutive ids.
+ *
+ * Relabeling is done inplace. The mapping is created from the union
+ * of the give arrays.
+ *
+ * \param arrays The id arrays to relabel.
+ * \return mapping array M from new id to old id.
+ */
+IdArray Relabel_(const std::vector<IdArray>& arrays);
```

Comment:
What's the behaviour when there are duplicate items in the input array?

Reply:
They will be mapped to the same ID. Added more description.